### PR TITLE
Fako/thumbnails

### PR DESCRIPTION
**Pull request by fako**
Implements creating previews through pipelines. This aims to improve the following:
* Cache thumbnail results so that thumbnails are not created every day (and potentially more frequent when pipelines run more often)
* Store errors that occur during thumbnail creation for easier debugging
* Share thumbnails between documents. Documents with identical URL's (but from different harvest versions) will share created thumbnails
* Remove thumbnails when they are stale. Thumbnails without any attached documents get deleted automatically.
* Makes thumbnails visible in the material sidebar (but only on acceptance for now)
* Allows syncing preview files on localhost with remote environment

Also: ExtructResource was not completely integrated yet and this PR does that.

Not included in this PR:
* PDF previews, that will be done in the next step. Tests for PDF previews break now, but keeping them around for future reference.
* HTML previews. Because cookie walls made this hard to do right it is unlikely to be implemented in the future.

### CODE REVIEW

File: portal/src/App.vue [hunk 1]
```diff
@@ -43,6 +43,18 @@ export default {
       this.$log.pageView(to.path)
       next()
     })
+  },
+  methods: {
+    isDemoEnvironment() {
+      if (this.$window.location.hostname.indexOf('acc.') >= 0) {
```
Comment (Ruben-Hartog) [inline portal/src/App.vue]: `return this.$window.location.hostname.indexOf('acc.') >= 0 || new URLSearchParams(this.$window.location.search).get('demo')`

File: service/surf/vendor/elasticsearch/serializers.py [hunk 2]
```diff
@@ -13,21 +14,23 @@
...
     Generate a presigned URL to share the S3 object where this resource is stored.
     If the application is not connected to S3 it simply returns a local path.
     """
-    if preview_path is None:
+    if url is None:
         return None
 
     if settings.AWS_HARVESTER_BUCKET_NAME is None:
-        return os.path.join(settings.MEDIA_URL, "harvester", preview_path, preview_type)
+        if "s3.amazonaws.com" not in url:
```
Comment (Ruben-Hartog) [inline service/surf/vendor/elasticsearch/serializers.py:26]: Is dit logica voor als je een dump restored hebt incl. previews?

**Review by Ruben-Hartog — APPROVED** [review]

File: service/surf/vendor/elasticsearch/serializers.py (hunk 2 above)
Comment (fako) [reply service/surf/vendor/elasticsearch/serializers.py:26]: Ja, precies. In de dump zitten de URL's van de environment waar je de dump van hebt. Dat betekent dus ook dat het lastiger is de dumps te delen tenzij de destination environment toegang heeft tot de source environment bucket. Maar momenteel zetten we datasets weinig tot nooit over tussen dev, acc en prod. Ik haal wel regelmatig een dataset op voor localhost. Daarvoor zit dit stukje erin.

File: portal/src/App.vue (hunk 1 above)
Comment (fako) [reply portal/src/App.vue]: Ok

**Pull request merged**